source/docs/flow_control.md

```diff
@@ -0,0 +1,103 @@
+### Overview
```

Comment:
Yep, this is what I was after. Some more suggestions on how to polish:
* Explain explicitly the connection between network and stream level handling (and how this relates to plain TCP or H1 network level handling).
* Any diagrams that are reasonable (i.e. don't take ages to put together) would help. I have in my mind the _perfect_ way to efficiently communicate this entire story, an animation showing what happens on backup, the series of firing callbacks and how the buffers fill and drain, but this would take a bit of time to put together, so I don't expect that from this PR (maybe for a future blog post however..).
* More explanation of who the principle actors are. Kind of like telling a story and providing some character background. Leading into the callback description, it would be good to know why HTTP connection manager and the router play the role they do, i.e. that they are the unique bridges across certain kinds of events and subsystem divides in Envoy.
* Cover corner cases explicitly, such as the combination of stream and connection level watermark firing.
* Run spell checker and fix any grammar nits.